plugins/web-plugin/src/main/java/ru/sbtqa/tag/pagefactory/web/utils/WebExpectedConditionsUtils.java

```diff
@@ -92,33 +93,45 @@ public static void waitForTextInInputExists(WebElement webElement, long timeout)
     /**
      * Wait until specified text either appears, or disappears from page source
      *
-     * @param text                text to search in page source
+     * @param text text to search in page source
      * @param shouldTextBePresent boolean, self explanatory
      * @throws ru.sbtqa.tag.pagefactory.exceptions.WaitException TODO
      */
     public static void waitForTextPresenceInPageSource(String text, boolean shouldTextBePresent) throws WaitException {
-        long timeoutTime = System.currentTimeMillis() + PageFactory.getTimeOut();
+        long timeoutTime = System.currentTimeMillis() + PROPERTIES.getTimeout() / 1000;
         WebElement body = waitUntilElementAppearsInDom(By.tagName("body"));
         while (timeoutTime > System.currentTimeMillis()) {
-            sleep(1);
             if (body.getText().replaceAll("\\s+", "").contains(text.replaceAll("\\s+", "")) == shouldTextBePresent) {
                 return;
             }
+            sleep(1);
+        }
+        throw new WaitException("Timed out after '" + PROPERTIES.getTimeout() + "' seconds waiting for presence of '" + text + "' in page source");
+    }
+
+    public static void waitForModalWindowWithText(String text) throws WaitException {
+        try {
+            String popupHandle = WebExpectedConditionsUtils.findNewWindowHandle((Set<String>) Stash.getValue("beforeClickHandles"));
+            if (null != popupHandle && !popupHandle.isEmpty()) {
+                PageContext.getCurrentPage().getDriver().switchTo().window(popupHandle);
+            }
+            waitForTextPresenceInPageSource(text, true);
+        } catch (Exception ex) {
+            throw new WaitException("Modal window with text '" + text + "' didn't appear during timeout", ex);
         }
-        throw new WaitException("Timed out after '" + PageFactory.getTimeOutInSeconds() + "' seconds waiting for presence of '" + text + "' in page source");
     }
 
     /**
      * @param existingHandles TODO
-     * @param timeout         TODO
+     * @param timeout TODO
      * @return TODO
      * @throws ru.sbtqa.tag.pagefactory.exceptions.WaitException TODO
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)